test/e2e/windows/cpu_limits.go

```diff
@@ -46,63 +46,61 @@ var _ = sigDescribe(feature.Windows, "Cpu Resources", framework.WithSerial(), sk
 	ginkgo.Context("Container limits", func() {
 		ginkgo.It("should not be exceeded after waiting 2 minutes", func(ctx context.Context) {
 			ginkgo.By("Creating one pod with limit set to '0.5'")
-			podsDecimal := newCPUBurnPods(1, powershellImage, "0.5", "1Gi")
-			e2epod.NewPodClient(f).CreateBatch(ctx, podsDecimal)
+			pods := newCPUBurnPods(1, powershellImage, "0.5", "1Gi")
+
 			ginkgo.By("Creating one pod with limit set to '500m'")
-			podsMilli := newCPUBurnPods(1, powershellImage, "500m", "1Gi")
-			e2epod.NewPodClient(f).CreateBatch(ctx, podsMilli)
-			ginkgo.By("Waiting 2 minutes")
-			time.Sleep(2 * time.Minute)
-			ginkgo.By("Ensuring pods are still running")
-			var allPods []*v1.Pod
-			for _, p := range podsDecimal {
-				pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(
-					ctx,
-					p.Name,
-					metav1.GetOptions{})
-				framework.ExpectNoError(err, "Error retrieving pod")
-				gomega.Expect(pod.Status.Phase).To(gomega.Equal(v1.PodRunning))
-				allPods = append(allPods, pod)
-			}
-			for _, p := range podsMilli {
-				pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(
-					ctx,
-					p.Name,
-					metav1.GetOptions{})
-				framework.ExpectNoError(err, "Error retrieving pod")
-				gomega.Expect(pod.Status.Phase).To(gomega.Equal(v1.PodRunning))
-				allPods = append(allPods, pod)
-			}
-			ginkgo.By("Ensuring cpu doesn't exceed limit by >5%")
-			for _, p := range allPods {
-				ginkgo.By("Gathering node summary stats")
-				nodeStats, err := e2ekubelet.GetStatsSummary(ctx, f.ClientSet, p.Spec.NodeName)
-				framework.ExpectNoError(err, "Error grabbing node summary stats")
-				found := false
-				cpuUsage := float64(0)
-				for _, pod := range nodeStats.Pods {
-					if pod.PodRef.Name != p.Name || pod.PodRef.Namespace != p.Namespace {
-						continue
+			pods = append(pods, newCPUBurnPods(1, powershellImage, "500m", "1Gi")...)
+			e2epod.NewPodClient(f).CreateBatch(ctx, pods)
+
+			ginkgo.By("Ensuring pods are running")
+			err := e2epod.WaitForPodsRunning(f.ClientSet, f.Namespace.Name, 2, framework.PodStartTimeout)
+			framework.ExpectNoError(err, "Error waiting for pods entering on running state.")
+
+			ginkgo.By("Gathering node summary stats")
+			err = wait.PollUntilContextTimeout(ctx, time.Minute*2, time.Minute*6, false, func(ctx context.Context) (bool, error) {
```

Comment:
could we add a note here about this not triggering until 2 mins go passed, then ever 2 mins?